test/net/sourceforge/kolmafia/session/ChoiceControlTest.java

```diff
@@ -366,6 +366,28 @@ void changingPlaqueRecordsNewName() {
     }
   }
 
+  @Nested
+  class Yachtzee {
+    @Test
+    void visitingTheNCResetsPreference() {
+      var cleanups = new Cleanups(withProperty("encountersUntilYachtzeeChoice ", 0));
+      try (cleanups) {
+        var req = new GenericRequest("adventure.php?snarfblat=404");
+        req.responseText = html("request/test_visit_yachtzee.html");
+        assertThat("encountersUntilYachtzeeChoice", isSetTo(20));
+      }
+    }
+
+    @Test
+    void choosingAnNCOptionDecrementsPreference() {
+      var cleanups =
+          new Cleanups(withProperty("encountersUntilYachtzeeChoice ", 20), withPostChoice2(918, 2));
```

Comment:
This feels like a counterintuitive logical flow. You're setting the `encountersUntilYachtzeeChoice` and processing the post-choice stuff (which presumably also modifies that preference) in the same `Cleanups`

look at how `changingPlaqueRecordsNewName`, right above this, works. That might help with both of these tests